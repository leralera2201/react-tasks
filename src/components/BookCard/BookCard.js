import React from "react";
import { Button, Card, CardBody, CardSubtitle, CardTitle } from "reactstrap";
import { useHistory } from "react-router-dom";

const BookCard = ({ book: { title, description, id } }) => {
  const history = useHistory();

  const goToDetail = () => {
    history.push(`/books/${id}`);
  };

  return (
    <Card className={"card-width"}>
      <CardBody>
        <CardTitle tag="h3">{title}</CardTitle>
        <CardSubtitle tag="h6" className="mb-2 text-muted mt-3">
          {description}
        </CardSubtitle>
        <Button onClick={goToDetail} className={"mt-3"}>
          Go to detail
        </Button>
      </CardBody>
    </Card>
  );
};

export default BookCard;
