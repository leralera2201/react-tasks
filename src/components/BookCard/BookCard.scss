.card-width {
  margin: 10px 0;
  flex: 0 1 23%;
}
