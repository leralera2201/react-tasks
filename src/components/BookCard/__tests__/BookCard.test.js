import React from 'react';
import { Router } from 'react-router-dom';
import {mount, shallow} from "enzyme";

import BookCard from "../BookCard";

describe("Testing BookCard component", () => {
    const book = {
        id: "1",
        title: "a",
        description: "a"
    }
    it("BookCard will render", () => {
        const wrapper = shallow(<BookCard book={book}/>);
        expect(wrapper.length).toEqual(1);
    })

    it("BookCard check goToDetail method", () => {
        const historyMock = { push: jest.fn(), location: {}, listen: jest.fn() };
        const wrapper = mount(<Router history={historyMock}><BookCard book={book}/></Router>)
        const button =wrapper.find('button');
        button.simulate('click');
        expect(historyMock.push.mock.calls[0][0]).toEqual(`/books/${book.id}`);
    })
})