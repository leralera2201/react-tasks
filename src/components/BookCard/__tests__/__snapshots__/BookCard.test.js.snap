// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing BookCard component renders correctly 1`] = `
<ContextProvider
  value={
    Object {
      "history": Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {},
        "push": [MockFunction],
      },
      "location": Object {},
      "match": Object {
        "isExact": false,
        "params": Object {},
        "path": "/",
        "url": "/",
      },
      "staticContext": undefined,
    }
  }
>
  <ContextProvider
    value={
      Object {
        "listen": [MockFunction] {
          "calls": Array [
            Array [
              [Function],
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        },
        "location": Object {},
        "push": [MockFunction],
      }
    }
  >
    <BookCard
      book={
        Object {
          "description": "a",
          "id": "1",
          "title": "a",
        }
      }
    />
  </ContextProvider>
</ContextProvider>
`;
